lib/client/client.js

```diff
@@ -185,13 +186,20 @@ Client.prototype.requestUpload = function (uri, file, callback) {
   } else {
     self.options.headers.Authorization = 'Basic ' + encoded;
   }
-
-  out = this._request(self.options, function (err, response, result) {
-    requestCallback(self, err, response, result, callback);
-  });
-
-  fs.createReadStream(file).pipe(out); // pipe the file!
-
+  self.emit('debug::request', self.options);
+  // assumes a binary file is passed in instead of a path
+  if(binary){
+    self.options.body = file;
+    return this._request(self.options, function (err, response, result) {
+      requestCallback(self, err, response, result, callback);
+    });
+  }
+  else{
+    let out = this._request(self.options, function (err, response, result) {
```

Comment:
`let` would mean everyone needs to be at Node 4.X or above. Can we assume that yet?